### PR TITLE
MethodHandle constants and invocation

### PR DESCRIPTION
**Pull request by akbertram**
This corrects the output of MethodHandle constants implemented in the previous pull request (#637) and adds support for MethodHandle.invoke() and MethodHandle.invokeExact(), which are signature polymorphic.

### CODE REVIEW

**Comment by akbertram** [comment]
@StevenArzt Anything missing from this PR? Happy to make any changes or improvements as requested.

File: src/soot/SootMethodRefImpl.java [hunk 1]
```diff
@@ -152,6 +163,11 @@ private SootMethod checkStatic(SootMethod ret) {
 	}
 
 	private SootMethod tryResolve(StringBuffer trace) {
+
+		if(isSignaturePolymorphic()) {
+			return new SootMethod(name(), parameterTypes(), returnType(), Modifier.PUBLIC);
```
Comment (StevenArzt) [inline src/soot/SootMethodRefImpl.java]: Please don't create SootMethods out of thin air. This method won't be registered in the Scene, will have no parent class, and will therefore violate various reasonable assumptions held by analysis clients.

**Comment by akbertram** [comment]
I've now ensured that the new SootMethod gets added to the Scene graph. Is there any way to test that this works correctly. Right now I run Soot's *intra*procedural optimization on a classfile that uses signature polymorphism, but that was passing before this change.

**Comment by mbenz89** [comment]
Can one of the admins verify this patch?

**Comment by akbertram** [comment]
Hi @StevenArzt, do these changes address your concerns or is there anything else I need to fix?

**Comment by ericbodden** [comment]
@StevenArzt could you please look into this again? Thanks!

**Comment by mbenz89** [comment]
ok to test

**Comment by mbenz89** [comment]
Signaturepolymorphic is now supported by #1135

**Pull request closed without merging**